packages/plugin-npm/sources/NpmSemverResolver.ts

```diff
@@ -70,6 +72,19 @@ export class NpmSemverResolver implements Resolver {
     const manifest = new Manifest();
     manifest.load(registryData.versions[version]);
 
+    // Manually add node-gyp dependency if there is a script using it and not already set
+    // This is because the npm registry will automatically add a `node-gyp rebuild` install script
+    // in the metadata if there is not already an install script and a binding.gyp file exists.
+    // Also, node-gyp is not always set as a dependency in packages, so it will also be added if used in scripts.
+    if (!manifest.dependencies.has(NODE_GYP_IDENT.identHash) || !manifest.peerDependencies.has(NODE_GYP_IDENT.identHash)) {
```

Comment:
Now that I think about it, shouldn't it rather be the following? As it stand it will be injected unless `node-gyp` is in both `dependencies` and `peerDependencies` 🤔 

```suggestion
    if (!manifest.dependencies.has(NODE_GYP_IDENT.identHash) && !manifest.peerDependencies.has(NODE_GYP_IDENT.identHash)) {
```

Reply:
Good catch!